### PR TITLE
Developer guide v1.2

### LINKED ISSUE #89: Update Developer Guide
Add service management use case in the DG.

### PR DESCRIPTION
**Pull request by galvinleow**
Check service management only, after concluding this function then I will use these to replicate success.

Closed #89

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-**Preconditions: Appointment list must exist**
+**Preconditions: Appointment list must exist.**
 
-**Guarantees: A new appointment will be added to the appointment list**
+**Guarantees: A new service will be added to the service list upon successful command.**
 
 **MSS**
+1. User adds a service by providing details.
+1. GrAB3 adds the service to the service list.  
+1. GrAB3 displays successful additional message.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: maybe can replace 'additional' with another word, or just write 'displays a success message'?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Preconditions: Service exists in service list.**
+
+**Guarantees: Updated service list with new service details.**
+
+**MSS**
+1. User requests to list all services.
+1. GrAB3 shows a list of services.
+1. User edits an existing service by providing the details.
+1. GrAB3 update the details of the selected service.
+1. GrAB3 update the service list.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *updates

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+Delete the selected service from the service list.
+
+**Preconditions: Service exists in service list.**
+
+**Guarantees: Updated service list with service specified by user deleted.**
+
+**MSS**
+1. User requests to list all services.
+1. GrAB3 shows a list of services.
+1. User request to delete a specific service.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *requests

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. GrAB3 delete a service.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *deletes

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+Find a service in the service list.
 
-    * 2a1. GrAB3 shows an error message.
+**Preconditions: Service exists in service list.**
 
-    Use case resumes at step 1.
+**Guarantees: Display all services in the service list that matches the search value.**
+
+**MSS**
+1. User requests to list all services that matches the search value.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *match

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. GrAB3 searches for services that matches search value.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *match

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. GrAB3 show list of all services that matches the search value.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *shows a, *match

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+Use case ends.
+
+**Extensions**
+* 2a. No service matches the search value.
+  * 2a1. GrAB3 displays empty list.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *an

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+#### UC005: List Services 
+List all the services in the service list.
+
+**Preconditions: Service exists in service list.**
+ 
+**Guarantees: All services in the service list will be displayed.**
+ 
+**MSS**
+1. User requests to list all services.
+1. GrAB3 display all services in the service list.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *displays

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+Use case ends.
+
+**Extensions**
+* 1a. No service in service list.
+  * 1a1. GrAB3 displays empty list.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *an

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  Use case resumes at step 1.
+  
+#### UC005: Clear Services
+Clear all services in the service list.
+
+**Preconditions: Service exists in service list**
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *At least one service exists in the service list

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Guarantees: An empty revenue list.**
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md:451]: Does it guarantee an empty service list as well?

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **MSS**
-
 1.  User requests to list all clients.
-2.  GrAB3 shows a list of clients.
-3.  User requests to edit the description of a specific client from that list.
-4.  GrAB3 shows the updated client.
-
-    Use case ends.
+1.  GrAB3 shows a list of clients.
+1.  User requests to edit the description of a specific client from that list.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: maybe can change description to details?

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-3.  User requests to sort the expenses by ascending order.
-4.  GrAB3 shows the sorted list view.
-
-    Use case ends.
+1.  User adds appointment into appointment list.
+1.  GrAB3 shows the updated list of appointments.
+Use case ends.
 
 **Extensions**
+* 2a. The specified module code is invalid.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *service

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -451,22 +523,20 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-1.  User requests to list all services.
-2.  GrAB3 shows a list of services.
-3.  User delete a service.
-4.  GrAB3 shows the updated list view.
-
-    Use case ends.
+1.  User requests to list all expenses.
+1.  GrAB3 shows a list of expenses.
+1.  User requests to sort the expenses by ascending order.
+1.  GrAB3 shows the sorted list view.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md]: *GrAB3 shows the list of expenses sorted in ascending order

**Review by yanlynnnnn — COMMENTED** [review]
👍🏻

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 (For all use cases below, the **System** is `GrAB3` and the **Actor** is the `user`, unless specified otherwise)
 
-### Appointment List
+### Services Management
 
-**UC01: Add an appointment into the appointment list**
+#### UC001: Add a Service 
+Add a service provided by the business into GrAB3.
```
Comment (hanstirtaputra) [inline docs/DeveloperGuide.md]: GrAB3 should be SuperSalon

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
```
Comment (hanstirtaputra) [inline docs/DeveloperGuide.md:331]: I think you can add **system** and **actor** here and for all other use cases

**Review by hanstirtaputra — CHANGES_REQUESTED** [review]
I think we can add System and Actor for each use case, and use this format consistently for all other use cases as well

File: docs/DeveloperGuide.md (hunk 12 above)
Comment (galvinleow) [reply docs/DeveloperGuide.md:451]: What you mean? If clear than should be empty?

File: docs/DeveloperGuide.md (hunk 13 above)
Comment (galvinleow) [reply docs/DeveloperGuide.md]: okay

File: docs/DeveloperGuide.md (hunk 14 above)
Comment (galvinleow) [reply docs/DeveloperGuide.md]: Will change this but likely this will not be in used because appointment use case not done

**Review by hanstirtaputra — APPROVED** [review]
LGTM

**Pull request merged**